crates/turbo-tasks-fs/src/glob.rs

```diff
@@ -457,19 +442,43 @@ mod tests {
         "node_modules/next/dist/server/next.js"
     )]
     #[case::node_modules_root("**/node_modules/**", "node_modules/next/dist/server/next.js")]
+    #[case::node_modules_root_package(
+        "**/node_modules/next/**",
+        "node_modules/next/dist/server/next.js"
+    )]
     #[case::node_modules_nested(
         "**/node_modules/**",
         "apps/some-app/node_modules/regenerate-unicode-properties/Script_Extensions/Osage.js"
     )]
+    #[case::node_modules_nested_package(
+        "**/node_modules/regenerate-unicode-properties/**",
+        "apps/some-app/node_modules/regenerate-unicode-properties/Script_Extensions/Osage.js"
+    )]
     #[case::node_modules_pnpm(
         "**/node_modules/**",
         "node_modules/.pnpm/regenerate-unicode-properties@9.0.0/node_modules/\
          regenerate-unicode-properties/Script_Extensions/Osage.js"
     )]
+    #[case::node_modules_pnpm_package(
+        "**/node_modules/{regenerate,regenerate-unicode-properties}/**",
+        "node_modules/.pnpm/regenerate-unicode-properties@9.0.0/node_modules/\
+         regenerate-unicode-properties/Script_Extensions/Osage.js"
+    )]
+    #[case::node_modules_pnpm_prefixed_package(
+        "**/node_modules/{@blockfrost/blockfrost-js,@highlight-run/node,@libsql/client,@jpg-store/\
+         lucid-cardano,@mikro-orm/core,@mikro-orm/knex,@prisma/client,@sentry/nextjs,@sentry/node,\
+         @swc/core,argon2,autoprefixer,bcrypt,better-sqlite3,canvas,cpu-features,cypress,eslint,\
+         express,next-seo,node-pty,payload,pg,playwright,postcss,prettier,prisma,puppeteer,rimraf,\
+         sharp,shiki,sqlite3,tailwindcss,ts-node,typescript,vscode-oniguruma,webpack,websocket,@\
+         aws-sdk/client-dynamodb,@aws-sdk/lib-dynamodb}/**",
+        "node_modules/.pnpm/@aws-sdk+lib-dynamodb@3.445.0_@aws-sdk+client-dynamodb@3.445.0/\
+         node_modules/@aws-sdk/lib-dynamodb/dist-es/index.js"
+    )]
     #[case::alternatives_nested1("{a,b/c,d/e/{f,g/h}}", "a")]
     #[case::alternatives_nested2("{a,b/c,d/e/{f,g/h}}", "b/c")]
     #[case::alternatives_nested3("{a,b/c,d/e/{f,g/h}}", "d/e/f")]
     #[case::alternatives_nested4("{a,b/c,d/e/{f,g/h}}", "d/e/g/h")]
+    // #[case::alternatives_chars("[abc]", "b")]
```

Comment:
is this inteded to be commented?

Reply:
This isn't supported in the glob parsing yet. We might want to uncomment that test case once we implement that.